accounting/aggregations/create_report.py

```diff
@@ -0,0 +1,214 @@
+    "--es-url-prefix": {"default" : "http://", "help" : "defaults to http:// or https:// with '--es-use-https'"},
+    "--es-index": {"default" : "chtc-schedd-*", "help" : "defaults to 'chtc-schedd-*'"},
+    "--es-user": {},
+    "--es-password-file": {"type": Path},
+    "--es-use-https": {"action": "store_true"},
+    "--es-ca-certs": {},
+    "--es-config-file": {
+        "type": Path,
+        "help": "JSON file containing an object that sets above ES options",
+    }
+}
+
+# =========== helper functions ===========
+
+def valid_date(date_str: str) -> datetime:
+    try:
+        return datetime.strptime(date_str, "%Y-%m-%d")
+    except ValueError:
+        raise argparse.ArgumentTypeError(f"Invalid date string, should match format YYYY-MM-DD: {date_str}")
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("--start", type=valid_date, required=True, 
+                        help="the date to start reporting on 'YYYY-MM-DD' (REQUIRED)")
+    parser.add_argument("--end", type=valid_date, default=datetime.strftime(datetime.now(), "%Y-%m-%d"),
+                        help="the date to end reporting on 'YYYY-MM-DD', defaults to the current date")
+
+    email_opts = parser.add_argument_group("email options")
+    for name, props in EMAIL_ARGS.items():
+        email_opts.add_argument(name, **props)
+
+    es_opts = parser.add_argument_group("elasticsearch options")
+    for name, props in ELASTICSEARCH_ARGS.items():
+        es_opts.add_argument(name, **props)
+
+    output_opts = parser.add_argument_group("output options")
+    for name, props in OUTPUT_ARGS.items():
+        output_opts.add_argument(name, **props)
+
+    return parser.parse_args()
+
+def get_client(args: dict) -> elasticsearch.Elasticsearch:
+    # set up elasticsearch options
+    es_opts = {
+        "timeout" : 120,
+        "hosts" : [args["es_url_prefix"] + args["es_host"]], 
+    }
+
+    if args["es_use_https"]:
+        if args["es_ca_certs"]:
+            es_opts.update({"ca_certs" : str(args["es_ca_certs"])})
+        
+        elif importlib.util.find_spec("certifi") is None:
+            print("error: using HTTPS requires either ca_certs or the certifi library to be installed")
+            sys.exit(1)
+
+        es_opts.update({"hosts" : "https://" + args["es_host"]})
+        es_opts.update({"use_https" : True})
+        es_opts.update({"verify_certs" : True}) 
+
+    if (not args["es_user"]) ^ (not args["es_password_file"]):
+        print("error: you must specify a username and password, or neither")
+        sys.exit(1)
+
+    elif args["es_user"] and args["es_password_file"]:
+        passwd_str = ""    
+        with open(args["es_password_file"], 'r') as file:
+           passwd_str = file.read()
+ 
+        es_opts.update({"http_auth" : (args["es_user"], passwd_str)})
+
+    client = elasticsearch.Elasticsearch(**es_opts)
+    connections.create_connection(alias="default", client=client)
```

Comment:
I see the Elasticsearch DSL docs suggest using this, but we also set `using=client` everywhere that it would matter (as far as I can tell). I hate to override what the docs say but I think it's better to be explicit about which Elasticsearch cluster you're using for every query, so I think we can just get rid of this (please test that this doesn't break anything, though).
```suggestion
```